x-pack/plugins/alerting/server/lib/get_rules_config.ts

```diff
@@ -6,23 +6,50 @@
  */
 
 import { omit } from 'lodash';
-import { RulesConfig, RuleTypeConfig } from '../config';
+import { Logger } from 'kibana/server';
+import { RulesConfig, RuleTypeConfig, RuleTypeConfigFromOriginPlugin } from '../config';
+import { validateDurationSchema } from '../../common';
+import { translations } from '../constants/translations';
+
+const DEFAULT_EXECUTION_TIMEOUT = '5m';
 
 export const getRulesConfig = ({
   config,
+  configFromOriginPlugin,
   ruleTypeId,
+  logger,
 }: {
   config: RulesConfig;
+  configFromOriginPlugin?: RuleTypeConfigFromOriginPlugin;
   ruleTypeId: string;
+  logger: Logger;
 }): RuleTypeConfig => {
   const ruleTypeConfig = config.execution.ruleTypeOverrides?.find(
     (ruleType) => ruleType.id === ruleTypeId
   );
 
+  // validate timeout from rule type registry here
+  let timeout = config.execution.timeout || DEFAULT_EXECUTION_TIMEOUT;
+
+  if (configFromOriginPlugin?.execution.timeout) {
```

Comment:
i moved the config validation here.
We were doing it in rule_type_registry and throw an error if the value is invalid.
But IMHO throwing error does not makes sense since the only way to get an invalid config is through the origin config and no end user can solve that problem.
With this new feature, if the value is invalid it fall-backs to the default value "5m" and logs an error.